apps/api/v1/lib/helpers/rateLimitApiKey.ts

```diff
@@ -6,15 +6,19 @@ export const rateLimitApiKey: NextMiddleware = async (req, res, next) => {
   if (!req.query.apiKey) return res.status(401).json({ message: "No apiKey provided" });
 
   // TODO: Add a way to add trusted api keys
-  await checkRateLimitAndThrowError({
-    identifier: req.query.apiKey as string,
-    rateLimitingType: "api",
-    onRateLimiterResponse: (response) => {
-      res.setHeader("X-RateLimit-Limit", response.limit);
-      res.setHeader("X-RateLimit-Remaining", response.remaining);
-      res.setHeader("X-RateLimit-Reset", response.reset);
-    },
-  });
+  try {
+    await checkRateLimitAndThrowError({
+      identifier: req.query.apiKey as string,
+      rateLimitingType: "api",
+      onRateLimiterResponse: (response) => {
+        res.setHeader("X-RateLimit-Limit", response.limit);
+        res.setHeader("X-RateLimit-Remaining", response.remaining);
+        res.setHeader("X-RateLimit-Reset", response.reset);
+      },
+    });
+  } catch (error) {
+    res.status(429).json({ message: "Rate limit exceeded" });
+  }
```

Comment:
no return here?

Reply:
isn't the next coming after enough ? once the status and json is set it should stay even after calling next()

Reply:
thinking about this, what's the outcome of catching the TRPC error sent by checkRateLimitAndThrowError and not throwing again afterward ? we might want to set status and json and then throw the error again

Reply:
Hmm will try that morgan

Reply:
Throwing an error in middle ware results in a 500 being thrown - Added tests this approach works as expected